### PR TITLE
drivers: Rename renesas ra drivers

### PR DESCRIPTION
**Pull request by aaillet**
With current name nomenclature for Renesas RA driver, it's not possible to target them correctly in the MAINTAINER file.

Rename all drivers to be able to delimit Renesas areas correctly.

Blocking #67487

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by bjarki-andreasen — DISMISSED** [review]

**Review by soburi — DISMISSED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
The Kconfig renames should be added to the migration guide/release notes.

**Comment by aaillet** [comment]
> The Kconfig renames should be added to the migration guide/release notes.

I added notes to both release note and migration guide, please @henrikbrixandersen tell me if I did it well !
Thanks

File: doc/releases/migration-guide-3.6.rst [hunk 1]
```diff
@@ -271,6 +271,13 @@ Device Drivers and Device Tree
   * ``DT_INST_IRQ_BY_IDX(inst, idx, irq)`` -> ``DT_INST_IRQN_BY_IDX(inst, idx)``
   * ``DT_INST_IRQ_BY_NAME(inst, name, irq)`` -> ``DT_INST_IRQN_BY_NAME(inst, name)``
 
+* Several Renesas RA series drivers Kconfig options have been renamed:
+
+  * :kconfig:option:`CLOCK_CONTROL_RA` -> :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
+  * :kconfig:option:`GPIO_RA` -> :kconfig:option:`GPIO_RENESAS_RA`
+  * :kconfig:option:`PINCTRL_RA` -> :kconfig:option:`PINCTRL_RENESAS_RA`
+  * :kconfig:option:`UART_RA` -> :kconfig:option:`UART_RENESAS_RA`
```
Comment (kartben) [inline doc/releases/migration-guide-3.6.rst:279]:
```suggestion
  * ``CLOCK_CONTROL_RA`` -> :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
  * ``GPIO_RA`` -> :kconfig:option:`GPIO_RENESAS_RA`
  * ``PINCTRL_RA`` -> :kconfig:option:`PINCTRL_RENESAS_RA`
  * ``UART_RA`` -> :kconfig:option:`UART_RENESAS_RA`
```

File: doc/releases/release-notes-3.6.rst [hunk 2]
```diff
@@ -157,6 +157,7 @@ Drivers and Sensors
 * Clock control
 
   * Renesas R-Car clock control driver now supports Gen4 SoCs
+  * Renamed :kconfig:option:`CLOCK_CONTROL_RA` with :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
```
Comment (kartben) [inline doc/releases/release-notes-3.6.rst:160]:
```suggestion
  * Renamed ``CLOCK_CONTROL_RA`` with :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
```

File: doc/releases/release-notes-3.6.rst [hunk 3]
```diff
@@ -179,6 +180,7 @@ Drivers and Sensors
 * GPIO
 
   * Renesas R-Car GPIO driver now supports Gen4 SoCs
+  * Renamed :kconfig:option:`GPIO_RA` with :kconfig:option:`GPIO_RENESAS_RA`
```
Comment (kartben) [inline doc/releases/release-notes-3.6.rst:183]:
```suggestion
  * Renamed ``GPIO_RA`` with :kconfig:option:`GPIO_RENESAS_RA`
```

File: doc/releases/release-notes-3.6.rst [hunk 4]
```diff
@@ -204,6 +206,7 @@ Drivers and Sensors
 * Pin control
 
   * Renesas R-Car pinctrl driver now supports Gen4 SoCs
+  * Renamed :kconfig:option:`PINCTRL_RA` with :kconfig:option:`PINCTRL_RENESAS_RA`
```
Comment (kartben) [inline doc/releases/release-notes-3.6.rst:209]:
```suggestion
  * Renamed ``PINCTRL_RA`` with :kconfig:option:`PINCTRL_RENESAS_RA`
```

File: doc/releases/release-notes-3.6.rst [hunk 5]
```diff
@@ -225,6 +228,8 @@ Drivers and Sensors
 
 * Serial
 
+  * Renamed :kconfig:option:`UART_RA` with :kconfig:option:`UART_RENESAS_RA`
```
Comment (kartben) [inline doc/releases/release-notes-3.6.rst:231]:
```suggestion
  * Renamed ``UART_RA`` with :kconfig:option:`UART_RENESAS_RA`
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by aaillet** [comment]
Addressed @kartben request

File: doc/releases/migration-guide-3.6.rst [hunk 6]
```diff
@@ -271,6 +271,13 @@ Device Drivers and Device Tree
   * ``DT_INST_IRQ_BY_IDX(inst, idx, irq)`` -> ``DT_INST_IRQN_BY_IDX(inst, idx)``
   * ``DT_INST_IRQ_BY_NAME(inst, name, irq)`` -> ``DT_INST_IRQN_BY_NAME(inst, name)``
 
+* Several Renesas RA series drivers Kconfig options have been renamed:
+
+  * ``CLOCK_CONTROL_RA`` -> :kconfig:option:`CLOCK_CONTROL_RENESAS_RA`
```
[the remark this answers is not in the document]
Comment (aaillet) [reply doc/releases/migration-guide-3.6.rst:276]: Thanks,
Don't I need the "config" prefix for previous names ?

[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply doc/releases/migration-guide-3.6.rst:276]: > Thanks, Don't I need the "config" prefix for previous names ? It feels weird for me not to use the same name nomenclature for old and new names

Yes, you need the `CONFIG_` prefix.

[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply doc/releases/migration-guide-3.6.rst:276]: > Just for you to know, I based these changed on https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-3.6.rst?plain=1#L361 So it looks like these rules are not widely known :)

Yeah, that change is wrong. Surprised it builds.

**Comment by aaillet** [comment]
Added "CONFIG" prefix to previous and new Kconfig options names in release note and migration guide

File: doc/releases/migration-guide-3.6.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (aaillet) [reply doc/releases/migration-guide-3.6.rst]: Thanks, Added config prefix.
If I'm understanding well, `:kconfig:option:` is checking for an existing Kconfig option at doc build time, that's it ?

[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply doc/releases/migration-guide-3.6.rst]: It's creating a link to it, yes - and we cannot create links to the now non-existing, previous names.

File: doc/releases/release-notes-3.6.rst [hunk 7]
```diff
@@ -157,6 +157,7 @@ Drivers and Sensors
 * Clock control
 
   * Renesas R-Car clock control driver now supports Gen4 SoCs
+  * Renamed ``CONFIG_CLOCK_CONTROL_RA`` with :kconfig:option:`CONFIG_CLOCK_CONTROL_RENESAS_RA`
```
[the remark this answers is not in the document]
Comment (aaillet) [reply doc/releases/release-notes-3.6.rst]: Fixed !

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**